src/core/Akka/Actor/ActorCell.Children.cs

```diff
@@ -54,7 +54,7 @@ private bool TryGetChild(string name, out IActorRef child)
 
         public virtual IActorRef AttachChild(Props props, bool isSystemService, string name = null)
         {
-            return ActorOf(props, name, true, isSystemService);
+            return MakeChild(props, name ?? GetRandomActorName(), true, isSystemService);
```

Comment:
But AttachChild is not related with ActorOf. Jvm version uses checkName in both of them